File: Easy/Second Largest/second-largest.cpp
//{ Driver Code Starts
#include <bits/stdc++.h>

using namespace std;

// } Driver Code Ends
//User function template for C++
class Solution{
public:	
	// Function returns the second
	// largest elements
	int print2largest(int arr[], int n) {
	    // code here
	   // M-1
	   sort(arr,arr+n);
	   int slargest = -1;
	   for(int i = n - 2; i >= 0; i--)
	   {
	       if(arr[i] != arr[n-1])
	       {
	           slargest = arr[i];
	           break;
	       }
	   }
	   
	   return slargest;
	}
};

//{ Driver Code Starts.

int main() {
    int t;
    cin >> t;
    while (t--) {
        int n;
        cin >> n;
        int arr[n];
        for (int i = 0; i < n; i++) {
            cin >> arr[i];
        }
        Solution ob;
        auto ans = ob.print2largest(arr, n);
        cout << ans << "\n";
    }
    return 0;
}

// } Driver Code Ends

File: Difficulty: Easy/Second Largest/second-largest.java
//{ Driver Code Starts
// Initial Template for Java
import java.io.*;
import java.util.*;
import java.util.stream.Collectors;

public class Main {

    public static void main(String[] args) throws Exception {
        Scanner sc = new Scanner(System.in);
        int t = Integer.parseInt(sc.nextLine());
        while (t-- > 0) {
            String[] arr1Str = sc.nextLine().split(" ");
            int[] arr = Arrays.stream(arr1Str).mapToInt(Integer::parseInt).toArray();
            Solution ob = new Solution();
            int ans = ob.getSecondLargest(arr);
            System.out.println(ans);

            System.out.println("~");
        }
    }
}

// } Driver Code Ends


// User function Template for Java

class Solution {
    public int getSecondLargest(int[] arr) {
        // Code Here
        int largest = arr[0];
        int sl = -1;
        
        for(int num : arr)
        {
            if(largest < num)
            {
                largest = num;
            }
            
        }
        
        for(int num : arr)
        {
            if(num != largest)
            {
                if(sl < num)
                {
                    sl = num;
                }
            }
        }
        
        return sl;
        
        
        
        
    }
}